### ISSUE TITLE
What is our default CI service?

### ISSUE THREAD

**Issue by purinkle** [thoughtbot/suspenders#1180]
With the introduction of #1172, we assumed that GitHub Actions is our default CI service. I love the new CI template. It's long overdue and something I've missed from Suspenders over the years. I would challenge the assumption that GitHub Actions is our default CI service. I lean towards CircleCI as a default. This is a personal preference, but we expect Suspenders to represent thoughtbot's opinion. What is thoughtbot's default CI service?

**Comment by stevepolitodesign** [comment]
Thank you for opening this. For context, I defaulted to GitHub Actions to [keep parity](https://github.com/rails/rails/pull/50508) with Rails in an effort to keep Suspenders lean and easier to maintain.

I think it would help to understand why we currently use CircleCI on other internal projects. Is it because, at one point, GitHub Actions either did not exist, or was not as effective as CircleCI? What does CircleCI offer that GitHub Actions does not?

One benefit to using CircleCI is that the repository does not need to be hosted on GitHub.

**Comment by ayellapragada** [comment]
The main reason for why we started standardizing around Github Actions for pipelines in general on Mission Control was that we found it much more likely for developers to investigate build failures when it was all in the same UI.

For example, when a non-test step would fail on another pipeline, such as AWS code build, or CircleCI, people tended to immediately ping the folks who created the pipelines, and asked them to take a look, even if the issue wasn't related to the pipeline, and was something that was in their change.

On the other hand, having it all in Github Actions, and in the same UI, made it significantly more likely that a developer would look at it, and probably resolve the issue on their own, without being gated by waiting on someone else, speeding up the feedback loop.

I don't have statistics or numbers to back this up, but that was the general feeling we noticed.

And granted that is more for continuous integration (build steps) and deployments, not CI in the sense of testing, which is what I think this issue is more about, but I think having a few things already in one place made it easier for other people to keep the tests in the same location as well.


And especially for things like dependecies for deployments, i.e. I want to push, run tests, then deploy if tests pass, is much simpler to do in one pipeline, rather than orchestrating between multiple.

**Comment by purinkle** [comment]
I agree. Suspenders should diverge from Rails as little as possible. This saves us time and effort with future work. Suspenders should also automate what we're doing instead of what we'd like. If we are using GitHub Actions as our current default, then let's reflect that here.

Currently, Suspenders has [a CI generator][] that generates a CircleCI configuration.

[The handbook][] also has explicit references to CircleCI. As does [the Playbook][].

The only reference I could find to the decision to [move to CircleCI][] is from 2015.

I have no strong opinions about what service we use. My only opinion is that Suspenders should reflect current thoughtbot practices.

[a ci generator]: https://github.com/thoughtbot/suspenders/blob/55cb5c246fe53aea4cf53436c7c208b40ad13e85/lib/suspenders/generators/ci_generator.rb
[the handbook]: https://github.com/thoughtbot/handbook/blob/4d98e03cb404f64c2bbc603dc2b47a8a17133664/business-operations/services/circle-ci.md
[the playbook]: https://github.com/thoughtbot/handbook/blob/4d98e03cb404f64c2bbc603dc2b47a8a17133664/playbook/developing/continuous-integration.md#L39
[move to circleci]: https://github.com/thoughtbot/suspenders/pull/592

**Comment by stevepolitodesign** [comment]
> [The handbook](https://github.com/thoughtbot/handbook/blob/4d98e03cb404f64c2bbc603dc2b47a8a17133664/business-operations/services/circle-ci.md) also has explicit references to CircleCI. As does [the Playbook](https://github.com/thoughtbot/handbook/blob/4d98e03cb404f64c2bbc603dc2b47a8a17133664/playbook/developing/continuous-integration.md#L39).
> 
> The only reference I could find to the decision to [move to CircleCI](https://github.com/thoughtbot/suspenders/pull/592) is from 2015.
> 

Good find! Thank you for sharing those. I think this issue could continue to serve as a conversation around this topic.

**Comment by purinkle** [comment]
Now that I've had a chance to sleep on these comments, I've had more thoughts.

How do we change a standard once we've set one?

Do we see where the prevailing wind of the team takes us and update things from there?

Does the existence of a standard hold us back from being more experimental?

These questions are outside this issue's scope, but I wanted to share them.

**Comment by mike-burns** [comment]
> How do we change a standard once we've set one?

Pull requests, either against the Playbook or, more practically, against Suspenders. Typically the Playbook PR comes after the Suspenders one.

**Comment by purinkle** [comment]
I had another "shower thought" about this subject today.

I looked to see what the Rails team's default CI service was.

Even before the recent changes, they used GitHub Actions.

If we wanted to keep parity, this would be a clear indicator.

The main sticking point for me is _why_ we use a particular service; apart from that, I have no strong opinions.

**Comment by stevepolitodesign** [comment]
@purinkle now that https://github.com/thoughtbot/handbook/pull/3804 has merged and Suspenders has been released, do you feel comfortable closing this issue?
